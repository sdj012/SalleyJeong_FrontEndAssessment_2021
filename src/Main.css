/* ===============
Styling Properties For All Components And Elements Nested In Main.js
=============== */

/* Test Complete Component */
.TestComplete-grid-container{
  background-color: #e1e1e1;
  padding-top:10%;
  padding-left: 10%;
  padding-right:10%;
  padding-bottom:10%;
}

  .complete-message{
    background-color:white;
    height:15vh;
    width:80vw;
    padding-top:2%;
    padding-bottom:2%;
    font-size: x-large;
    color:rgb(20, 20, 139);
    font-weight: bolder;
    border-radius: 25px;
  }

  .complete-message > button{
    border:none;
    height:45px;
    width:100px;
    background-color:#388e3c;
    font-size:large;
    font-weight: bolder;

  }
  

/* Outer Wrapper */
.home-grid-container{
  padding-top:10%;
  padding-left: 10%;
  padding-right:10%;
  padding-bottom:10%;
  background-color: #e1e1e1;
}



/* Inner Wrapper */

.main-grid-container{
  border-radius: 25px;
  height:100%;
  width:100%;

  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr 1fr;
  gap: 0px 0px;
  grid-auto-flow: row;
  grid-template-areas:
    "."
    ".";
  
  background-color: white;
}



/* Wrapper - Area A */

.area-A{
  border-radius: 25px;
  padding-top:5%;
  align-self: center;
  justify-self: center;
  height:50vh;
  width:80vw;
  background-color:white;
  line-height: 8vh;
}

  /* Blurb Component */

  .Blurb-grid-container{
    font-size: x-large;
    color:rgb(0, 0, 0);
  }

  /* Scramble Component */

  .Scramble-grid-container{
    font-size:50px;
    color:rgb(27, 27, 167);
    font-weight: bolder;
    padding-bottom:3%;
  }



/* Wrapper - Area B */

.area-B{
  border-radius: 25px;
  align-self: center;
  justify-self: center;
  height:50vh;
  width:80vw;
  background-color:white;
  margin-bottom:5%;
}


  /* ContainerComponent */

  .container{
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
  }

  /* Score */
  .score{
    font-size:x-large;
    font-weight:bolder;
  }

/* Next Button */
.next{
  background-color:#388e3c;
}


/* Row Component */

.RowComponent{
  display:flex;
  flex-direction: row;
  justify-content: flex-start;
}


/* Column Component */

.column-style{
  width: 100%;
  height:40%;
  border:none;
  margin: 5px 5px 5px 5px;
}

.column-style > input{  
  padding-left:50%;
  height:70px;
  color:white;
  font-size:xx-large;
}

button{
  border:none;
  height:45px;
  width:100px;
  margin-top:10%;
  color:white;
  background-color:#388e3c;
  font-size:xx-large;
  font-weight: bolder;
}


