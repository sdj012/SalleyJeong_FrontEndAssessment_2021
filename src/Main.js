import React, { Component } from 'react';
import axios from 'axios';
import './Main.css';
import ContainerComponent from './ContainerComponent';
import Blurb from './Blurb';
import TestComplete from './TestComplete';

class Main extends React.Component {

  constructor(props){

    super(props);

    this.state={ 
      pagination:1,
      sentence:'',
      score:0,
      scrambled:'',
      scrambledMiddle:'',
      gameComplete:false,
      markBlockComplete:false,
    }
  
    this.getData=this.getData.bind(this);
    this.setSentence=this.setSentence.bind(this);
    this.returnSentence=this.returnSentence.bind(this);
    this.markBlockComplete=this.markBlockComplete.bind(this);

  }

  setSentence=(data)=>{

    this.setState({
      sentence:data,
    })

  }


  returnSentence=()=>{ 
    return this.state.sentence;
  }

  paginate=(e)=>{
    
    e.preventDefault();
    this.setState({
      pagination:this.state.pagination+1,
      markBlockComplete:false,
    },function(){
      this.getData();
    })

  }

  getData=()=>{
    axios.get("https://api.hatchways.io/assessment/sentences/" + this.state.pagination)
    .then(response => this.setSentence(response["data"]["data"]["sentence"]));
  }

  componentDidMount=()=>{
    this.getData();
  }


  markBlockComplete=(isComplete)=>{

    if(isComplete){

      this.setState({
        markBlockComplete:true,
        score:this.state.score+1
      })

    }

  }

  render() {

    if(this.state.score<10){

      return(

        <div className="main-grid-container">
    
            <div className="area-A">

              <Blurb key={this.state.sentence} sentence={this.state.sentence}/>
              
            </div>

            <div className="area-B">

              <ContainerComponent key={this.state.sentence} sentence={this.state.sentence} markBlockComplete={this.markBlockComplete}/>

              <button hidden={!this.state.markBlockComplete} onClick={this.paginate}>Next</button>

            </div>


        </div>

      )
    }

    else {
      return(
        <div><TestComplete/></div>
      )
    }
  }

}

export default Main;