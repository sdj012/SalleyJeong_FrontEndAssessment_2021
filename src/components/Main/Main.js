import React from 'react';
import axios from 'axios';
import './Main.css';
import Blurb from './Blurb/Blurb';
import BlockComponent from './Block/BlockComponent';
import TestComplete from './TestComplete/TestComplete';

class Main extends React.Component {
  
  constructor(props){

    super(props);

    this.state={ 
      pagination:1,
      numberOfSentences:10,
      sentence:"",
      score:0,
      markBlockComplete:false,
    }

    this.getData=this.getData.bind(this);
    this.setSentence=this.setSentence.bind(this);
    this.markBlockComplete=this.markBlockComplete.bind(this);
    this.detectEnter=this.detectEnter.bind(this);
  }

  componentDidMount=()=>{
    this.getData();
  }

  getData=()=>{

    axios.get("https://api.hatchways.io/assessment/sentences/" + this.state.pagination)
    .then(response => this.setSentence(response["data"]["data"]["sentence"]));
  }

  setSentence=(data)=>{

    data=data.toLowerCase()

    this.setState({
      sentence:data,
    })

  }

  paginate=(e)=>{
    
    e.preventDefault();

    this.setState({
      pagination:this.state.pagination+1,
      markBlockComplete:false,
      score:this.state.score+1,
    },function(){
      this.getData();
    })

  }

  detectEnter=(event)=>{

    event.preventDefault();

    if(event.keyCode === 13 )this.paginate(event);

  }

  /* Receives Data OnCompleteness From Child Component and Communicates To State To Update 'Score' */

  markBlockComplete=(isComplete)=>{

    if(isComplete){

      this.setState({
        markBlockComplete:true,
      })

    }

  }
  
  render() {

    /* If Game Is Complete: Render Test Complete Component*/

    if(this.state.score === this.state.numberOfSentences){ 

      return(

        <div>
          <TestComplete/>
        </div>

      )

    }

    else { /* Else: Render Quizzes Generated From Fetched Data*/

      return(

        <div className="home-grid-container" data-testid="main-test">

        <div className="main-grid-container">

          {/* Renders: Scrambled Sentence and Current Score*/}

            <div className="area-A"> 

              <Blurb key={this.state.sentence} sentence={this.state.sentence}/>

              <div className="score"> Score: {this.state.score}</div>
              
            </div>

          {/* Renders: Series of Text Inputs and Conditionally Visible 'Next' Button */}

            <div className="area-B"> 

              <BlockComponent key={this.state.sentence} sentence={this.state.sentence} markBlockComplete={this.markBlockComplete} callPaginate={this.paginate}/>

            </div>

        </div>

      </div>
      )
    }
  }

}

export default Main;