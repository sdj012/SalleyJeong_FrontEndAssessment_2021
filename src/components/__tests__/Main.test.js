import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Main from '../Main/Main';

afterEach(()=>{
  cleanup();
});

test('should render Main component', () => {
  const main = { pagination:1, numberOfSentences:'10', sentence:"i love learning code", score:0, markBlockComplete:false}
  render (<Main main={main}/>);
  const MainElement = screen.getByTestId('main-test');
  expect(MainElement).toBeInTheDocument();
  expect(MainElement).toHaveTextContent('Score')
});

test('matches snapshot',()=>{
  const main = { pagination:1, numberOfSentences:'10', sentence:"i love learning code", score:0, markBlockComplete:false}
  const tree = renderer.create(<Main main={main}/>).toJSON()
  expect(tree).toMatchSnapshot();
})
