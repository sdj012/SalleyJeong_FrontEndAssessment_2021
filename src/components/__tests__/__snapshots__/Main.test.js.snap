// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="home-grid-container"
  data-testid="main-test"
>
  <div
    className="main-grid-container"
  >
    <div
      className="area-A"
    >
      <div
        className="Blurb-grid-container"
      >
        <div
          className="Scramble-grid-container"
          id="scrambled-word"
        >
           
        </div>
        <div>
          Guess the sentence! Start Typing
        </div>
        <div>
          The yellow blocks are meant for spaces
        </div>
      </div>
      <div
        className="score"
      >
         Score: 
        0
      </div>
    </div>
    <div
      className="area-B"
    >
      <div
        className="container"
        data-testid="block-test"
      >
        <div
          className="RowComponent"
        />
        <div
          hidden={true}
        >
          <div
            className="NextButton-grid-container"
          >
            <button
              className="next"
              onClick={[Function]}
              onKeyDown={[Function]}
            >
              Next
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
